lib/src/feature_file.dart

```diff
@@ -62,10 +63,19 @@ class FeatureFile {
         _testerType,
         _testerName,
         isIntegrationTest,
+        hookFile,
       );
 
   List<StepFile> getStepFiles() => _stepFiles;
 
+  HookFile? get hookFile => generatorOptions.enableHooks
+      ? HookFile.create(
+          featureDir: featureDir,
+          package: package,
+          generatorOptions: generatorOptions,
+        )
+      : null;
```

Comment:
With this approach, the `hookFile` property will be calculated each time it is accessed. Please make it a property
```dart
final HookFile? hookFile;
```
and initialize it in the constructor, like:
```dart
  })  : _lines = _prepareLines(
          input.split('\n').map((line) => line.trim()).map(BddLine.new),
        ),
        hookFile = generatorOptions.enableHooks
            ? HookFile.create(
                featureDir: featureDir,
                package: package,
                generatorOptions: generatorOptions,
              )
            : null {
```